oot.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/bash
+
+while getopts ':c:' OPT; do
+        case ${OPT} in
+        c)
+		case ${OPTARG} in
+		clang|gcc|linaro_gcc)
+			export _compiler=${OPTARG}
+			;;
+		*)
+			echo "Compiler '${OPTARG}' unknown or not implemented"
+			exit
+			;;
+		esac
+		;;
+        ?)
+                echo && echo "Incorrect option: ${OPT}"
+                exit
+                ;;
+        esac
+done
+
+shift $(( OPTIND - 1 ))
+
+
+if (( "$#" == 0 )); then
+	echo "Usage: $0 <device> (Example for akatsuki: $0 akatsuki)"
+	exit
+fi
+
+
+# Default to clang
+if [[ -z "${_compiler}" ]]; then
+	export _compiler=clang
+fi
+
+
+# Platform common
+export BOARD_KERNEL_BASE=0x00000000
+export BOARD_KERNEL_PAGESIZE=4096
+export BOARD_KERNEL_TAGS_OFFSET=0x01E00000
+export BOARD_RAMDISK_OFFSET=0x02000000
+
+for _device in "$@"; do
+	unset BOARD_KERNEL_CMDLINE _platform _has_dtbo _recovery_ramdisk
+
+	BOARD_KERNEL_CMDLINE+=" lpm_levels.sleep_disabled=1"
+	BOARD_KERNEL_CMDLINE+=" service_locator.enable=1"
+
+	# Device specific
+	case ${_device} in
+	mermaid)
+		export _platform=ganges
+		;;
+	discovery|pioneer|voyager)
+		export _platform=nile
+		;;
+	bahamut|griffin)
+		export _platform=kumano
+		;;
+	pdx201)
+		export _platform=seine
+		;;
+	akari|akatsuki|apollo)
+		export _platform=tama
+		;;
+	lilac|maple|poplar)
+		export _platform=yoshino
+		;;
+	*)
+		echo "Device '${_device}' unknown or not implemented"
+		exit
+		;;
+	esac
+
+
+	# Platform specific
+	case ${_platform} in
+#	nile|ganges)
+#		_verity_file=build/target/product/security/verity.x509.pem
+#		_verity_key_id=$(openssl x509 -in $_verity_file -text | grep keyid | sed 's/://g' | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]' | sed 's/keyid//g')
+#
+#		BOARD_KERNEL_CMDLINE+=" veritykeyid=id:$_verity_key_id"
+#		export BOARD_KERNEL_CMDLINE
+#		;;
```

Comment:
I don't want to loose verity on these platforms, shall I test this locally and uncomment the case?

While at it we should move `ANDROID_ROOT` into this file and change it to 

```bash
		_verity_file=$ANDROID_ROOT/build/target/product/security/verity.x509.pem
```

Reply:
I've done it because I don't see ir anywehere in SODP trees:

```
$ grep -r BOARD_KERNEL_CMDLINE device/sony/ | grep veritykeyid
$
```

Reply:
[For posterity] `veritykeyid` is set here: https://cs.android.com/android/platform/superproject/+/master:build/make/core/Makefile;l=742?q=VERITY_KEYID